chat_v2_message_counter_sample/src/main/java/com/zendesk/chat_v2/sample/MainActivity.java

```diff
@@ -0,0 +1,55 @@
+package com.zendesk.chat_v2.sample;
+
+import android.os.Bundle;
+import android.view.View;
+import android.widget.TextView;
+
+import androidx.appcompat.app.AppCompatActivity;
+
+import zendesk.chat.ChatConfiguration;
+import zendesk.chat.ChatEngine;
+import zendesk.messaging.MessagingActivity;
+
+public class MainActivity extends AppCompatActivity implements UnreadMessageCounter.UnreadMessageCounterListener {
+
+    private UnreadMessageCounter unreadMessageCounter;
+    private TextView messageCounterView;
+
+    @Override
+    protected void onCreate(Bundle savedInstanceState) {
+        super.onCreate(savedInstanceState);
+
+        if (Global.isMissingCredentials()) {
+            setContentView(R.layout.missing_credentials);
+            return;
+        }
+        setContentView(R.layout.activity_main);
+
+        messageCounterView = findViewById(R.id.tvMessageCounter);
+        unreadMessageCounter = new UnreadMessageCounter(this);
```

Comment:
This doesn't actually need to be a field, we could just make it 
```suggestion
        final UnreadMessageCounter unreadMessageCounter = new UnreadMessageCounter(this);
```